nanoFramework.Device.Bluetooth/BluetoothLEDevice.cs

```diff
@@ -315,36 +324,67 @@ internal GattReadResult ReadAttributeValue(ushort AttributeHandle)
             GattCommunicationStatus status = GattCommunicationStatus.Unreachable;
             Buffer buffer = null;
             ushort result = 0;
+            bool securityStarted = false;
+            bool retryRead;
 
             lock (lockObj)
             {
-                _completedEvent.Reset();
-
-                //Debug.WriteLine($"# ReadValue start con:{ConnectionHandle} attrHandle:{AttributeHandle}  ");
-                result = NativeStartReadValue(ConnectionHandle, AttributeHandle);
-                if (result == 0)
+                do
                 {
-                    // Read started, wait for completed event
-                    if (_completedEvent.WaitOne(operationTimeout, false))
+                    retryRead = false;
+
+                    //Debug.WriteLine($"# ReadAttributeValue");
```

Comment:
either keep the Debug.WriteLine and uncomment, either remove it fully. Same in other places. There is no issue keeping them uncommented as they'll get removed in Released

Reply:
The reason I leave them commented is so I can selectively uncomment during debugging a problem. If all uncommented, then too much debug info.  Later will remove them.

Reply:
added enum